### PR TITLE
New pull request for Tf-Idf work.

### PR DESCRIPTION
**Pull request by rawoke083**
Think I screwed up some rebasing/merging. Making a new pull request for Tf-Idf work :S Sorry for the hassle.

### CODE REVIEW

**Comment by rawoke083** [comment]
@jbrukh Conflicts resolved :)

File: bayesian.go [hunk 1]
```diff
@@ -254,6 +324,38 @@ func (c *Classifier) Learn(document []string, which Class) {
 	c.learned++
 }
 
+// Here we use all the TF samples for the class and convert
+// them to TF-IDF https://en.wikipedia.org/wiki/Tf%E2%80%93idf
+// once we have finished learning all the classes and have the totals.
+func (c *Classifier) ConvertTermsFreqToTfIdf() {
+
+	if c.DidConvertTfIdf {
+		panic("TfIdf-Mode:Cumulative counts - can only call this once. Reset and relearn.")
```
Comment (jbrukh) [inline bayesian.go]: Last nitpicks, I promise! :)

"Cannot call ConvertTermsFreqToTfIdf more than once. Reset and relearn to reconvert."

File: bayesian.go [hunk 2]
```diff
@@ -274,6 +376,10 @@ func (c *Classifier) Learn(document []string, which Class) {
 // Unlike c.Probabilities(), this function is not prone to
 // floating point underflow and is relatively safe to use.
 func (c *Classifier) LogScores(document []string) (scores []float64, inx int, strict bool) {
+	if c.tfIdf && !c.DidConvertTfIdf {
+		panic("Error:Using TF-IDF classifier:Need to call ConvertTermsFreqToTfIdf's first ")
```
Comment (jbrukh) [inline bayesian.go]: "Using a TF-IDF classifier. Please call ConvertTermsFreqToTfIdf before calling LogScores."

File: bayesian.go [hunk 3]
```diff
@@ -341,6 +450,10 @@ func (c *Classifier) ProbScores(doc []string) (scores []float64, inx int, strict
 // Underflow detection is more costly because it also
 // has to make additional log score calculations.
 func (c *Classifier) SafeProbScores(doc []string) (scores []float64, inx int, strict bool, err error) {
+	if c.tfIdf && !c.DidConvertTfIdf {
+		panic("Using a TF-IDF classifier. Please call ConvertTermsFreqToTfIdf before calling LogScores.")
```
Comment (jbrukh) [inline bayesian.go]: "SafeProbScores", not "LogScores".

**Comment by jbrukh** [comment]
Last nitpicks and we're ready to merge.

**Comment by rawoke083** [comment]
@jbrukh - Fixed. No worries :) Should have caught most of those anyway !

**Comment by jbrukh** [comment]
Merged! Thanks for the patch.

**Comment by rawoke083** [comment]
:) Thanks for patience !!

**Pull request merged**